source/lac/trilinos_vector.cc

```diff
@@ -215,13 +239,20 @@ namespace TrilinosWrappers
     void
     Vector::reinit (const IndexSet &parallel_partitioner,
                     const MPI_Comm &communicator,
-                    const bool      omit_zeroing_entries)
+                    const bool      /*omit_zeroing_entries*/)
     {
       nonlocal_vector.reset();
 
       Epetra_Map map = parallel_partitioner.make_trilinos_map (communicator,
                                                                true);
-      reinit (map, omit_zeroing_entries);
+
+      vector.reset (new Epetra_FEVector(map));
+
+      has_ghosts = vector->Map().UniqueGIDs()==false;
+
+      owned_elements = parallel_partitioner;
```

Comment:
What happens if somebody provides overlapping sets to this function? ``owned_elements`` will be wrong here and we can not decide who the owner should be. Maybe we should assert ``has_ghosts==false``?